packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -45,6 +52,11 @@ void updateXcodeGeneratedProperties(String projectPath, BuildMode mode, String t
     localsBuffer.writeln('LOCAL_ENGINE=${localEngineArtifacts.engineOutPath}');
   }
 
+  // If the project isn't using platform plugins, don't assume CocoaPods will be
+  // needed and depended on.
+  if (hasPlugins)
+    localsBuffer.writeln('#include "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"');
```

Comment:
ya, I noticed that too. Though it seems like we hardcode the Release configuration in the xcodebuild command anyway
https://github.com/flutter/flutter/blob/b8b6a14d8b1629b24278a01e3dd5ffc06f0756f4/packages/flutter_tools/lib/src/ios/mac.dart#L166